apps/barista-design-system/src/pages/single-page/components/toc/toc.ts

```diff
@@ -87,23 +87,15 @@ export class BaToc implements OnInit, AfterViewInit, OnDestroy {
   }
 
   /** @internal handle the click on a TOC item */
-  _handleTocClick(ev: MouseEvent): void {
-    /* Preventing the default behavior is necessary, because on Angular component pages
-     * there's a base URL defined and the on-page-links are always relative to "/"
-     * and not to the current page. */
-    ev.preventDefault();
-    ev.stopImmediatePropagation();
-    const targetId = (ev.currentTarget as HTMLElement).getAttribute('href');
-    const target = this._document.querySelector(targetId || '');
-
-    if (this._platform.isBrowser && target) {
-      // Has to be set manually because of preventDefault() above.
-      window.location.hash = targetId || '';
+  _handleTocClick(target: string): void {
+    this._location.onUrlChange(() => {
```

Comment:
You are adding this listener on every click. I don't think we need that one at all. Does location.go not scroll to the correct anchored position already?
It might have been thrown off by the duplicate ids on the page, see comments above.

Reply:
I removed the listener and the scroll animation. `location.go()` does not scroll. Also not stated like that in the documentation.
FYI: Changes the browser's URL to a normalized version of a given URL, and pushes a new item onto the platform's history.